### PR TITLE
Restoring multiple times with same RestoreCommand now clears summary …

### PR DESCRIPTION
**Pull request by BrennanConroy**
…messages

### CODE REVIEW

**Comment by davidfowl** [comment]
Can you show before and after since there are no tests

**Comment by BrennanConroy** [comment]
Before:
multirestore
After:
Same except remove the 10 package(s) line

Easily reproducible by installing a global command.

**Comment by davidfowl** [comment]
What happens if you pass multiple projects to restore?

**Comment by BrennanConroy** [comment]
Nothing different.

**Comment by davidfowl** [comment]
Here's what @BrennanConroy and I decided, add a small `RestoreMessageContext` that has Errors and Info messages and pass that to each invoke call. That state is different per call to Execute()

File: src/Microsoft.Framework.PackageManager/Restore/RestoreCommand.cs [hunk 1]
```diff
@@ -71,8 +67,33 @@ public RestoreCommand(FrameworkName fallbackFramework)
         public bool SkipRestoreEvents { get; set; }
         public bool IgnoreMissingDependencies { get; set; }
 
-        private Dictionary<string, List<string>> ErrorMessages { get; set; }
-        private Dictionary<string, List<string>> InformationMessages { get; set; }
+        private class SummaryContext
```
Comment (davidfowl) [inline src/Microsoft.Framework.PackageManager/Restore/RestoreCommand.cs]: Private classes always go to the bottom of the file.

File: src/Microsoft.Framework.PackageManager/Restore/RestoreCommand.cs [hunk 2]
```diff
@@ -211,10 +216,10 @@ public RestoreCommand(FrameworkName fallbackFramework)
                     Reports.Information.WriteLine(string.Format("Total time {0}ms", sw.ElapsedMilliseconds));
                 }
 
-                if (_installCount > 0)
+                if (summary.InstallCount > 0)
                 {
-                    InformationMessages.GetOrAdd("Installed:", _ => new List<string>()).Add($"{_installCount} package(s) to {packagesDirectory}");
-                    _installCount = 0;
+                    summary.InformationMessages.GetOrAdd("Installed:", _ => new List<string>()).Add($"{summary.InstallCount} package(s) to {packagesDirectory}");
+                    summary.InstallCount = 0;
```
Comment (davidfowl) [inline src/Microsoft.Framework.PackageManager/Restore/RestoreCommand.cs]: Why do you reset it?

**Comment by davidfowl** [comment]
@BrennanConroy What's the deal with this change?

File: src/Microsoft.Framework.PackageManager/Restore/RestoreCommand.cs [hunk 3]
```diff
@@ -85,46 +106,20 @@ public RestoreCommand(FrameworkName fallbackFramework)
                 RestoreDirectories.Add(Directory.GetCurrentDirectory());
             }
 
+            var summary = new SummaryContext();
+
             bool success = true;
             foreach (var dir in RestoreDirectories.Select(Path.GetFullPath).Distinct())
             {
-                success &= await Execute(dir);
+                success &= await Execute(dir, summary);
```
Comment (troydai) [inline src/Microsoft.Framework.PackageManager/Restore/RestoreCommand.cs]: Could there be a race condition?

Comment (troydai) [inline src/Microsoft.Framework.PackageManager/Restore/RestoreCommand.cs]: Never mind, saw the `GetOrAdd` extension.

**Comment by davidfowl** [comment]
@BrennanConroy Is this going to make it for beta7?

**Comment by BrennanConroy** [comment]
Yes!

File: src/Microsoft.Framework.PackageManager/Restore/RestoreCommand.cs (hunk 2 above)
Comment (BrennanConroy) [inline src/Microsoft.Framework.PackageManager/Restore/RestoreCommand.cs]: Removing

File: src/Microsoft.Framework.PackageManager/Restore/RestoreCommand.cs [hunk 4]
```diff
@@ -533,6 +509,7 @@ public RestoreCommand(FrameworkName fallbackFramework)
             if (!SkipInstall)
             {
                 await InstallPackages(installItems, packagesDirectory);
+                summary.InstallCount += installItems.Count();
```
Comment (davidfowl) [inline src/Microsoft.Framework.PackageManager/Restore/RestoreCommand.cs]: `.Count` not `.Count()`

**Comment by davidfowl** [comment]
:shipit: After that small change

**Comment by BrennanConroy** [comment]
f27d11518c78009eb44e6985e343e69efbb430ee

**Pull request closed without merging**